caravel/assets/javascripts/SqlLab/components/SouthPane.jsx

```diff
@@ -1,25 +1,42 @@
 import { Alert, Tab, Tabs } from 'react-bootstrap';
 import QueryHistory from './QueryHistory';
 import ResultSet from './ResultSet';
-import { areArraysShallowEqual } from '../../reduxUtils';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as Actions from '../actions';
 import React from 'react';
 
 import shortid from 'shortid';
 
+// editorQueries are queries executed by users
+// passed from SqlEditor component
+// queries are all queries executed
+// passed from global state for the purpose of data preview tabs
 const propTypes = {
-  queries: React.PropTypes.array.isRequired,
+  editorQueries: React.PropTypes.array.isRequired,
+  queries: React.PropTypes.object.isRequired,
   actions: React.PropTypes.object.isRequired,
+  dataPreviewQueryIds: React.PropTypes.array,
+  activeSouthPaneTab: React.PropTypes.string,
+};
+
+const defaultProps = {
+  dataPreviewQueryIds: [],
+  activeSouthPaneTab: 'Results',
 };
 
 class SouthPane extends React.PureComponent {
-  shouldComponentUpdate(nextProps) {
```

Comment:
We could still put a shouldComponentUpdate here deeply comparing whether queries, editorQueries, activeSouthPaneTab and dataPreviewQueryIds changed, but I'm not sure whether it's worth it to do a comparison on all queries just to prevent re-rendering and if it actually saves us time since queries object are fairly large. Any suggestions? @mistercrunch